src/pytest_subtests/plugin.py

```diff
@@ -122,10 +142,27 @@ def _addSubTest(
                 node=self, call=call_info, report=sub_report
             )
 
+        # Add non-subtest skips once all subtest failures are processed by # `_addSubTest`.
+        non_subtest_skip = [
+            (x, y)
+            for x, y in self.instance._outcome.skipped
+            if not isinstance(x, _SubTest)
+        ]
+        subtest_errors = [
+            (x, y) for x, y in self.instance._outcome.errors if isinstance(x, _SubTest)
+        ]
+        if len(subtest_errors) > 0 and len(non_subtest_skip) > 0:
+            last_subset_error = subtest_errors[-1]
+            if exc_info is last_subset_error[-1]:
+                for testcase, reason in non_subtest_skip:
+                    self._originaladdSkip(testcase, reason)  # type: ignore[attr-defined]
+
 
 def pytest_configure(config: pytest.Config) -> None:
     TestCaseFunction.addSubTest = _addSubTest  # type: ignore[attr-defined]
     TestCaseFunction.failfast = False  # type: ignore[attr-defined]
+    TestCaseFunction._originaladdSkip = copy.copy(TestCaseFunction.addSkip)  # type: ignore[attr-defined]
+    TestCaseFunction.addSkip = _addSkip  # type: ignore[method-assign]
```

Comment:
The `_addSkip` in `unittest.case` doesn't contain logic to call `addSubTest` for `outcome.skipped` (unlike `_feedErrorsToResult`), and it turns out this causes the reporting issue described in the PR.

Unlike `addSubTest` (where original `TestCaseFunction` doesn't have it), `addSkip` is defined `TestCaseFunction`, and here we save the original implementation and patch `TestCaseFunction.addSkip` to a newly defined `_addSkip` that allows us to call `addSubTest`.

The `_originaladdSkip` allows us to restore to it in `pytest_unconfigure` as well as it turns out necessary to call it too inside the newly defined `_addSkip`.